### PR TITLE
Configuration property for clients idle timeout

### PR DESCRIPTION
**Pull request by pv3nturi**
New config property for clients idle timeout: _**clients.idle.timeout**_

### CODE REVIEW

File: carapace-server/src/main/java/org/carapaceproxy/server/ClientConnectionHandler.java [hunk 1]
```diff
@@ -132,6 +134,17 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
         LOG.log(Level.SEVERE, "bad error", cause);
     }
 
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+        if (evt instanceof IdleStateEvent) {
+            IdleStateEvent e = (IdleStateEvent) evt;
+            if (e.state() == IdleState.ALL_IDLE) {
+                LOG.log(Level.SEVERE, "Idle connection detected for client {0}. Channel closed.", new Object[]{clientAddress});
```
Comment (hamadodene) [inline carapace-server/src/main/java/org/carapaceproxy/server/ClientConnectionHandler.java:142]: Maybe debug?

**Review by hamadodene — APPROVED** [review]
LGTM

File: carapace-server/src/test/java/org/carapaceproxy/RawClientTest.java [hunk 2]
```diff
@@ -565,6 +565,49 @@ public void testRequestsReadTimeout() throws Exception {
...
+            RuntimeServerConfiguration currentConfiguration = server.getCurrentConfiguration();
+            currentConfiguration.setClientsIdleTimeoutSeconds(30);
+            server.getConnectionsManager().applyNewConfiguration(currentConfiguration);
+
+            stats = server.getConnectionsManager().getStats();
+
+            try (RawHttpClient client = new RawHttpClient("localhost", port)) {
+                for (int j = 0; j < 2; j++) {
+                    RawHttpClient.HttpResponse res = client.get("/index.html");
+                    Thread.sleep(60_000);
```
Comment (nicoloboschi) [inline carapace-server/src/test/java/org/carapaceproxy/RawClientTest.java:595]: maybe it is better to change the new parameter with a lower value and reduce test duration?

File: carapace-server/src/main/java/org/carapaceproxy/server/ClientConnectionHandler.java (hunk 1 above)
Comment (nicoloboschi) [reply carapace-server/src/main/java/org/carapaceproxy/server/ClientConnectionHandler.java:142]: +1 for debug level

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

**Review by nicoloboschi — APPROVED** [review]

**Pull request merged**